app/screens/post_options/bindings/bindings.tsx

```diff
@@ -76,9 +77,10 @@ type OptionProps = {
     },
 }
 
-const Option = injectIntl((props: OptionProps) => {
-    const onPress = async () => {
-        const {closeWithAnimation, post, teamID} = props;
+class Option extends React.PureComponent<OptionProps> {
+    onPress = async () => {
```

Comment:
I wonder why not keep both components using the same approach (submitting state, do not close in error, etc...).

Reply:
@larkox I talked with @abhijit-singh and this is where we landed (having the post menu close right away). Questioning it again, I actually think having the same behavior among the components may be better.

@abhijit-singh's point was that the channel info screen is its own page, so exiting that is more significant than hiding the post menu. What do you guys think?

Reply:
I like coherence and homogeneity, and that is why I am rooting for having the same behaviour, but 0/5.

Reply:
@larkox @mickmister  So we currently have different behaviours for the two menus. Don't think we should change that only for apps.  For instance, clicking on `Edit Channel` in the Channel Menu opens a form and then brings the user back to the channel menu on Save. 

In the Post action menu, clicking on something like `Edit`, first closes the post action menu and then brings the user to the post editing UI. And clicking on Save from there takes the user back to the center channel. 

I know we're talking about error states here and I tried to produce them by turning off the internet connection. They seem to have different behaviours currently as well, so I feel we should keep it that way unless we feel strongly about this and are willing to make that change for all kinds of actions, not just for apps.